cli/cmd/generate_gcp.go

```diff
@@ -0,0 +1,780 @@
+			}
+
+			if GenerateGcpCommandState.OrganizationIntegration {
+				mods = append(mods, gcp.WithOrganizationIntegration(GenerateGcpCommandState.OrganizationIntegration))
+			}
+
+			if GenerateGcpCommandState.EnableForceDestroyBucket {
+				mods = append(mods, gcp.WithEnableForceDestroyBucket())
+			}
+
+			if GenerateGcpCommandState.EnableUBLA {
+				mods = append(mods, gcp.WithEnableUBLA())
+			}
+
+			// Create new struct
+			data := gcp.NewTerraform(
+				GenerateGcpCommandState.Config,
+				GenerateGcpCommandState.AuditLog,
+				mods...)
+
+			// Generate
+			hcl, err := data.Generate()
+			cli.StopProgress()
+
+			if err != nil {
+				return errors.Wrap(err, "failed to generate terraform code")
+			}
+
+			// Write-out generated code to location specified
+			dirname, location, err := writeGeneratedCodeToLocation(cmd, hcl, "gcp")
+			if err != nil {
+				return err
+			}
+
+			// Prompt to execute
+			err = SurveyQuestionInteractiveOnly(SurveyQuestionWithValidationArgs{
+				Prompt:   &survey.Confirm{Default: GenerateGcpCommandExtraState.TerraformApply, Message: QuestionRunTfPlan},
+				Response: &GenerateGcpCommandExtraState.TerraformApply,
+			})
+
+			if err != nil {
+				return errors.Wrap(err, "failed to prompt for terraform execution")
+			}
+
+			// Execute
+			locationDir := filepath.Dir(location)
+			if GenerateGcpCommandExtraState.TerraformApply {
+				// Execution pre-run check
+				err := executionPreRunChecks(dirname, locationDir)
+				if err != nil {
+					return err
+				}
+			}
+
+			// Output where code was generated
+			if !GenerateGcpCommandExtraState.TerraformApply {
+				cli.OutputHuman(provideGuidanceAfterExit(false, false, locationDir, "terraform"))
+			}
+
+			return nil
+		},
+		PreRunE: func(cmd *cobra.Command, _ []string) error {
+			// Validate output location is OK if supplied
+			dirname, err := cmd.Flags().GetString("output")
+			if err != nil {
+				return errors.Wrap(err, "failed to load command flags")
+			}
+			if err := validateOutputLocation(dirname); err != nil {
+				return err
+			}
+
+			//// set to nil if LogBucketLifecycleRuleAge value is -9999
+			//// -9999 is a dummy default value used to verify if the user has provided a value
+			//bucketLifecycleRuleAge, err := cmd.Flags().GetInt("bucket_lifecycle_rule_age")
+			//if err != nil {
+			//	return errors.Wrap(err, "failed to load command flags")
+			//}
+			//
+			//if bucketLifecycleRuleAge == 99999 && *GenerateGcpCommandState.LogBucketLifecycleRuleAge == 99999 {
+			//	nilSetter(GenerateGcpCommandState.LogBucketLifecycleRuleAge)
+			//}
+
+			// Validate gcp sa credentials file, if passed
+			gcpSaCredentials, err := cmd.Flags().GetString("service_account_credentials")
+			if err != nil {
+				return errors.Wrap(err, "failed to load command flags")
+			}
+			if err := validateServiceAccountCredentialsFile(gcpSaCredentials); gcpSaCredentials != "" && err != nil {
```

Comment:
could `validateServiceAccountCredentialsFile` be updated to just return nil if `credFile` is ""?